core/src/main/scala/kafka/log/Log.scala

```diff
@@ -553,6 +553,14 @@ class Log(@volatile var logStartOffset: Long,
 
   /** Only used for ZK clusters when we update and start using topic IDs on existing topics */
   def assignTopicId(topicId: Uuid): Unit = {
+    // defensively check that any newly assign topic ID matches any that is already set
+    _topicId.foreach { current =>
+      if (!current.equals(topicId))
```

Comment:
Can we shortcut return if the current topicId is already defined and matches the provided topicId?

Reply:
We can. I was just looking at this case again.